src/parselmouth/PointProcess.cpp

```diff
@@ -0,0 +1,306 @@
+
+	// DIRECT (INTEGER_PointProcess_getNumberOfPoints) {
+	def("get_number_of_points", [](PointProcess self) { return self->nt; });
+
+	// FORM (INTEGER_PointProcess_getNumberOfPeriods, U"PointProcess: Get number of periods", U"PointProcess: Get number of periods...") {
+	def("get_number_of_periods", PointProcess_getNumberOfPeriods, GET_RANGE_DEFAULT_PROPERTIES);
+
+	// FORM (REAL_PointProcess_getTimeFromIndex, U"Get time", 0 /*"PointProcess: Get time from index..."*/) {
+	def(
+		"get_time_from_index", [](PointProcess self, int pointNumber) {
+			return (pointNumber <= 0 || pointNumber > self->nt) ? py::none() : py::cast(self->t[pointNumber]);
+		},
+		"point_number"_a, "get time associated with the point number (1-based index)");
+
+	// DIRECT(HELP_PointProcess_help){
+	// DIRECT(PLAY_PointProcess_hum)
+
+	// FORM (MODIFY_PointProcess_addPoint, U"PointProcess: Add point", U"PointProcess: Add point...") {
+	def("add_point", &PointProcess_addPoint, "time"_a = 0.5);
+
+	// FORM (MODIFY_PointProcess_addPoints, U"PointProcess: Add points", U"PointProcess: Add point...") {
+	// def("add_points", &PointProcess_addPoints, "times"_a = std::vector<double>({0.1, 0.2, 0.5}));
+
+	// DIRECT(NEW1_PointProcesses_intersection)
+	def("intersection", &PointProcesses_intersection, "other"_a);
+
+	// DIRECT(PLAY_PointProcess_play)
+
+	// FORM(MODIFY_PointProcess_removePoint, U"PointProcess: Remove point", U"PointProcess: Remove point...")
+	def("remove_point", &PointProcess_removePoint, "point_number"_a = 1);
+
+	// FORM(MODIFY_PointProcess_removePointNear, U"PointProcess: Remove point near", U"PointProcess: Remove point near...")
+	def("remove_point_near", &PointProcess_removePointNear, "time"_a = 0.5);
+
+	// FORM(MODIFY_PointProcess_removePoints, U"PointProcess: Remove points", U"PointProcess: Remove points...")
+	def("remove_points", &PointProcess_removePoints, "from_point_number"_a = 1, "to_point_number"_a = 10);
+
+	// FORM(MODIFY_PointProcess_removePointsBetween, U"PointProcess: Remove points between", U"PointProcess: Remove points between...")
+	def("remove_points_between", &PointProcess_removePointsBetween, "from_time"_a = 0.3, "to_time"_a = 0.7);
+
+	// FORM(MODIFY_PointProcess_voice, U"PointProcess: Fill unvoiced parts", nullptr)
+	def("voice", &PointProcess_voice, "period"_a = 0.01, "maximum_voiced_period"_a = 0.02000000001);
+
+	// non-Script PointProcess APIs
+	def(
+		"get_window_points", [](PointProcess self, double tmin, double tmax) {
+			const MelderIntegerRange points = PointProcess_getWindowPoints(self, tmin, tmax);
+			return std::make_tuple(points.first, points.last);
+		},
+		"tmin"_a, "tmax"_a);
+
+	// DIRECT(NEW1_PointProcesses_difference)
+	def("difference", &PointProcesses_difference, "other"_a.none(false));
+
+	// DIRECT(NEW1_PointProcesses_union)
+	def("union", &PointProcesses_union, "other"_a);
+
+	// Skip? FORM(GRAPHICS_PointProcess_draw, U"PointProcess: Draw", nullptr)
```

Comment:
That would be all Python solution wouldn't it? My instinct tells me to keep that project separate, call it "Parselmouth-GUI" or something like that, which depends on Parselmouth. That way, users like me (who don't care for Praat UI) can use Parselmouth without baggage, especially when building an executable. (Now, here is a very good reason for you to move the project to the Parselmouth organization!)

Reply:
> That would be all Python solution wouldn't it?

Not fully, because in the ideal case, I'd have Praat's internal drawing functions call back into Python's matplotlib or so. It wouldn't really be a GUI just like a library like seaborn (extending matplotlib and making new fancy drawings) wouldn't really be a GUI.